offline_imu_cam/plugin.cpp

```diff
@@ -21,9 +21,20 @@ class offline_imu_cam : public ILLIXR::threadloop {
 	virtual skip_option _p_should_skip() override {
 		if (_m_sensor_data_it != _m_sensor_data.end()) {
 			dataset_now = _m_sensor_data_it->first;
-			reliable_sleep(std::chrono::nanoseconds{dataset_now - dataset_first_time} + real_first_time);
-			real_now = real_first_time + std::chrono::nanoseconds{dataset_now - dataset_first_time};
-			return skip_option::run;
+			// Sleep for the difference between the current IMU vs 1st IMU and current UNIX time vs UNIX time the component was init
+			std::this_thread::sleep_for(
+				std::chrono::nanoseconds{dataset_now - dataset_first_time}
+				+ real_first_time
+				- std::chrono::high_resolution_clock::now()
+			);
```

Comment:
Is this due to the time reference (dataset vs. real) conversion? If so, can you explain more in the comment?